segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/BookKeeperLog.java

```diff
@@ -400,37 +400,39 @@ private boolean processPendingWrites() {
             toExecute = this.writes.getWritesToExecute(maxTotalSize);
         }
 
-        // Execute the writes.
-        log.debug("{}: Executing {} writes.", this.traceObjectId, toExecute.size());
-        for (int i = 0; i < toExecute.size(); i++) {
-            Write w = toExecute.get(i);
-            try {
-                // Record the beginning of a new attempt.
-                int attemptCount = w.beginAttempt();
-                if (attemptCount > this.config.getMaxWriteAttempts()) {
-                    // Retried too many times.
-                    throw new RetriesExhaustedException(w.getFailureCause());
-                }
-
-                // Invoke the BookKeeper write.
-                w.getWriteLedger().ledger.asyncAddEntry(w.data.array(), w.data.arrayOffset(), w.data.getLength(), this::addCallback, w);
-            } catch (Throwable ex) {
-                // Synchronous failure (or RetriesExhausted). Fail current write.
-                boolean isFinal = !isRetryable(ex);
-                w.fail(ex, isFinal);
-
-                // And fail all remaining writes as well.
-                for (int j = i + 1; j < toExecute.size(); j++) {
-                    toExecute.get(j).fail(new DurableDataLogException("Previous write failed.", ex), isFinal);
+        if (toExecute.size() > 0) {
```

Comment:
It might be simpler if this were inverted, so it could be handled as a special case where it just returned true.

Reply:
That's a good suggestion. It turns out if I do that I'll have too many `return` statements in this method so Checkstyle won't be happy.

I refactored the method and extracted some code out into two different methods.